src/frequenz/sdk/timeseries/_ringbuffer.py

```diff
@@ -76,8 +76,8 @@ def __init__(
         self._time_index_alignment: datetime = time_index_alignment
 
         self._gaps: list[Gap] = []
-        self._datetime_newest: datetime = datetime.min
-        self._datetime_oldest: datetime = datetime.max
+        self._datetime_newest: datetime = datetime.min.replace(tzinfo=timezone.utc)
+        self._datetime_oldest: datetime = datetime.max.replace(tzinfo=timezone.utc)
```

Comment:
Here `datetime.max` (and later `datetime.min` are used just as markers, right? I would define them as class constants for 2 reasons: We can change the way we represent the markers, and we don't need to do the `.replace()` every time we want to use them:

```py
class OrderedRingBuffer:
    _DATETIME_MIN = datetime.min.replace(tzinfo=timezone.utc)
    _DATETIME_MAX = datetime.max.replace(tzinfo=timezone.utc)
    
    def __init__(self, ...):
        ...
        self._datetime_newest: datetime = self._DATETIME_MIN
        self._datetime_oldest: datetime = self._DATETIME_MAX
```

In the future I would replace that with a sentinel as IMHO it is even more clear that it is used as a marker. With the above we could even use `is not` to check if it is the marker instead of `!=`. But in this case is not as straightforward because if we really hit the min/max datetime in the value of `self._datetime_oldest` or `self._datetime_newest` then we are screwed :D

Reply:
Yes, they are markers. I'll change it as suggested in a separate commit.